### PR TITLE
listen for SIGTERM and trigger shutdown

### PR DESCRIPTION
**Pull request by bbalser**

### CODE REVIEW

File: ingest/src/main.rs [hunk 1]
```diff
@@ -51,9 +51,12 @@ impl Server {
         poc_metrics::start_metrics(&settings.metrics)?;
 
         let (shutdown_trigger, shutdown_listener) = triggered::trigger();
+        let mut sigterm = signal::unix::signal(signal::unix::SignalKind::terminate())?;
```
Comment (jeffgrunewald) [inline ingest/src/main.rs:54]: does this need to be called outside of the spawned task for any reason? just to catch and propagate the error if it occurs?

**Review by jeffgrunewald — APPROVED** [review]

File: ingest/src/main.rs (hunk 1 above)
Comment (bbalser) [reply ingest/src/main.rs:54]: just to propagate the error, rather than unwrapping in the tokio::spawn call.

**Comment by andymck** [comment]
Tested locally and will still sometimes hit the following filestore error when ctrl c on the iot verifier...Whether directly related or not I am not entirely sure

2023-04-28T08:19:11.674801Z  INFO iot_verifier::loader: processing 9 ingest files of type iot_beacon_ingest_report
^C2023-04-28T08:19:12.572633Z  INFO file_store::file_sink: stopping file sink iot_invalid_witness
2023-04-28T08:19:12.572642Z  INFO file_store::file_sink: stopping file sink iot_invalid_witness
2023-04-28T08:19:12.572636Z  INFO file_store::file_sink: stopping file sink iot_invalid_beacon
2023-04-28T08:19:12.572653Z  INFO file_store::file_info_poller: FileInfoPoller shutting down
2023-04-28T08:19:12.572653Z  INFO file_store::file_info_poller: FileInfoPoller shutting down
2023-04-28T08:19:12.572680Z  INFO file_store::file_upload: stopping file uploader
2023-04-28T08:19:12.572657Z  INFO file_store::file_sink: stopping file sink iot_poc
2023-04-28T08:19:12.572704Z  INFO file_store::file_sink: stopping file sink iot_invalid_beacon
2023-04-28T08:19:12.573137Z  INFO iot_verifier::entropy_loader: stopping verifier entropy_loader
2023-04-28T08:19:12.573185Z  INFO iot_verifier::packet_loader: stopping verifier iot packet loader
2023-04-28T08:19:12.573192Z  INFO iot_verifier::purger: stopping purger
2023-04-28T08:19:12.573203Z  INFO file_store::file_sink: stopping file sink iot_reward_share
2023-04-28T08:19:12.573205Z  INFO file_store::file_sink: stopping file sink reward_manifest
2023-04-28T08:19:12.573207Z  INFO file_store::file_upload: stopping file uploader
2023-04-28T08:19:12.578146Z ERROR file_store::file_sink: file_sink write failed for "iot-verifier_valid_poc" with SendError(Data(Sender { inner: Some(Inner { state: State { is_complete: false, is_closed: false, is_rx_task_set: false, is_tx_task_set: false } }) }, [10, 32, 41, 186, 55, 109, 234, 13, 206, 5, 103, 103, 84, 200, 205, 63, 176, 96, 128, 186, 103, 129, 152, 62, 138, 215, 68, 232, 138, 117, 242, 45, 106, 39, 18, 255, 1, 8, 142, 151, 158, 182, 252, 48, 18, 18, 54, 51, 49, 48, 52, 53, 50, 48, 57, 55, 48, 49, 53, 48, 53, 48, 50, 51, 24, 203, 17, 34, 218, 1, 18, 33, 0, 49, 207, 3, 57, 59, 45, 249, 54, 35, 12, 43, 83, 29, 252, 231, 30, 211, 103, 5, 1, 147, 60, 37, ......]))
2023-04-28T08:19:12.580570Z ERROR iot_verifier::runner: failed to save invalid_witness_report to s3, mpsc channel error
2023-04-28T08:19:12.580683Z ERROR file_store::file_sink: file_sink write failed for "iot-verifier_valid_poc" with SendError(Data(Sender { inner: Some(Inner { state: State { is_complete: false, is_closed: false, is_rx_task_set: false, is_tx_task_set: false } }) }, [10, 32, 42, 122, 143, 179, 151, 85, 117, 12, 56, 255, 9, 130, 126, 65, 74, 170, 0, 51, 191, 44, 34, 170, 71, 214, 200, 241, 42, 216, 1, 77, 254, 93, 18, 254, 1, 8, 149, 171, 158, 182, 252, 48, 18, 18, 54, 51, 49, 48, 53, 51, 55, 53, 48, 50, 50, 48, 52, 54, 51, 49, 48, 51, 24, 144, 78, 34, 218, 1, 18, 33, 0, 230, ....]))
2023-04-28T08:19:12.580802Z ERROR iot_verifier::runner: failed to save invalid_witness_report to s3, mpsc channel error
2023-04-28T08:19:12.581351Z ERROR file_store::file_sink: file_sink write failed for "iot-verifier_valid_poc" with SendError(Data(Sender { inner: Some(Inner { state: State { is_complete: false, is_closed: false, is_rx_task_set: false, is_tx_task_set: false } }) }, [10, 32, 133, 221, 32, 158, 52, 41, 54, 207, 228, 236, 235, 210, 227, 64, 186, 32, 255, 214, 100, 125, 42, 251, 112, 98, 200, 170, 230, 60, 84, 167, 174, 1, 18, 134, 2, 8, 227, .....]))
2023-04-28T08:19:12.581417Z ERROR iot_verifier::runner: failed to save invalid_witness_report to s3, mpsc channel error

**Comment by bbalser** [comment]
> Tested locally and will still sometimes hit the following filestore error when ctrl c on the iot verifier...Whether directly related or not I am not entirely sure
> 
> 2023-04-28T08:19:11.674801Z INFO iot_verifier::loader: processing 9 ingest files of type iot_beacon_ingest_report ^C2023-04-28T08:19:12.572633Z INFO file_store::file_sink: stopping file sink iot_invalid_witness 2023-04-28T08:19:12.572642Z INFO file_store::file_sink: stopping file sink iot_invalid_witness 2023-04-28T08:19:12.572636Z INFO file_store::file_sink: stopping file sink iot_invalid_beacon 2023-04-28T08:19:12.572653Z INFO file_store::file_info_poller: FileInfoPoller shutting down 2023-04-28T08:19:12.572653Z INFO file_store::file_info_poller: FileInfoPoller shutting down 2023-04-28T08:19:12.572680Z INFO file_store::file_upload: stopping file uploader 2023-04-28T08:19:12.572657Z INFO file_store::file_sink: stopping file sink iot_poc 2023-04-28T08:19:12.572704Z INFO file_store::file_sink: stopping file sink iot_invalid_beacon 2023-04-28T08:19:12.573137Z INFO iot_verifier::entropy_loader: stopping verifier entropy_loader 2023-04-28T08:19:12.573185Z INFO iot_verifier::packet_loader: stopping verifier iot packet loader 2023-04-28T08:19:12.573192Z INFO iot_verifier::purger: stopping purger 2023-04-28T08:19:12.573203Z INFO file_store::file_sink: stopping file sink iot_reward_share 2023-04-28T08:19:12.573205Z INFO file_store::file_sink: stopping file sink reward_manifest 2023-04-28T08:19:12.573207Z INFO file_store::file_upload: stopping file uploader 2023-04-28T08:19:12.578146Z ERROR file_store::file_sink: file_sink write failed for "iot-verifier_valid_poc" with SendError(Data(Sender { inner: Some(Inner { state: State { is_complete: false, is_closed: false, is_rx_task_set: false, is_tx_task_set: false } }) }, [10, 32, 41, 186, 55, 109, 234, 13, 206, 5, 103, 103, 84, 200, 205, 63, 176, 96, 128, 186, 103, 129, 152, 62, 138, 215, 68, 232, 138, 117, 242, 45, 106, 39, 18, 255, 1, 8, 142, 151, 158, 182, 252, 48, 18, 18, 54, 51, 49, 48, 52, 53, 50, 48, 57, 55, 48, 49, 53, 48, 53, 48, 50, 51, 24, 203, 17, 34, 218, 1, 18, 33, 0, 49, 207, 3, 57, 59, 45, 249, 54, 35, 12, 43, 83, 29, 252, 231, 30, 211, 103, 5, 1, 147, 60, 37, ......])) 2023-04-28T08:19:12.580570Z ERROR iot_verifier:🏃 failed to save invalid_witness_report to s3, mpsc channel error 2023-04-28T08:19:12.580683Z ERROR file_store::file_sink: file_sink write failed for "iot-verifier_valid_poc" with SendError(Data(Sender { inner: Some(Inner { state: State { is_complete: false, is_closed: false, is_rx_task_set: false, is_tx_task_set: false } }) }, [10, 32, 42, 122, 143, 179, 151, 85, 117, 12, 56, 255, 9, 130, 126, 65, 74, 170, 0, 51, 191, 44, 34, 170, 71, 214, 200, 241, 42, 216, 1, 77, 254, 93, 18, 254, 1, 8, 149, 171, 158, 182, 252, 48, 18, 18, 54, 51, 49, 48, 53, 51, 55, 53, 48, 50, 50, 48, 52, 54, 51, 49, 48, 51, 24, 144, 78, 34, 218, 1, 18, 33, 0, 230, ....])) 2023-04-28T08:19:12.580802Z ERROR iot_verifier:🏃 failed to save invalid_witness_report to s3, mpsc channel error 2023-04-28T08:19:12.581351Z ERROR file_store::file_sink: file_sink write failed for "iot-verifier_valid_poc" with SendError(Data(Sender { inner: Some(Inner { state: State { is_complete: false, is_closed: false, is_rx_task_set: false, is_tx_task_set: false } }) }, [10, 32, 133, 221, 32, 158, 52, 41, 54, 207, 228, 236, 235, 210, 227, 64, 186, 32, 255, 214, 100, 125, 42, 251, 112, 98, 200, 170, 230, 60, 84, 167, 174, 1, 18, 134, 2, 8, 227, .....])) 2023-04-28T08:19:12.581417Z ERROR iot_verifier:🏃 failed to save invalid_witness_report to s3, mpsc channel error

This is because the file_sink is shutting down before the main app tasks get shutdown.  Gonna need to refactor file_sink to fix this.  I will start working on that, but lets get this PR in as is.

**Review by andymck — APPROVED** [review]

**Pull request merged**